src/main/java/at/laborg/briss/BrissGUI.java

```diff
@@ -404,80 +385,54 @@ private void setWorkingState(String stateMessage) {
-        previewPanel.removeAll();
-        progressBar.setString(Messages.getString("BrissGUI.reloadingFile")); //$NON-NLS-1$
-        ClusterPagesTask clusterTask = new ClusterPagesTask(workingSet.getSourceFile(), getExcludedPages());
+        ClusterPagesTask clusterTask = new ClusterPagesTask(loadFile, password, null);
         clusterTask.addPropertyChangeListener(this);
         clusterTask.execute();
     }
 
-    private void maximizeWidthInSelectedRects() {
-        // maximize to width
-        // search for maximum width
-        int maxWidth = -1;
-        for (MergedPanel panel : mergedPanels) {
-            int panelMaxWidth = panel.getWidestSelectedRect();
-            if (maxWidth < panelMaxWidth) {
-                maxWidth = panelMaxWidth;
-            }
-        }
-        // set maximum width to all rectangles
-        if (maxWidth == -1)
-            return;
-        for (MergedPanel mp : mergedPanels) {
-            mp.setSelCropWidth(maxWidth);
-        }
-    }
+    private String promptForPassword() {
+        JPanel panel = new JPanel();
 
-    private void maximizeHeightInSelectedRects() {
-        // maximize to height
-        // search for maximum height
-        int maxHeight = -1;
-        for (MergedPanel panel : mergedPanels) {
-            int panelMaxHeight = panel.getHighestSelectedRect();
-            if (maxHeight < panelMaxHeight) {
-                maxHeight = panelMaxHeight;
-            }
-        }
-        // set maximum height to all rectangles
-        if (maxHeight == -1)
-            return;
-        for (MergedPanel mp : mergedPanels) {
-            mp.setSelCropHeight(maxHeight);
-        }
-    }
+        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
 
-    private void maximizeSizeInAllRects() {
-        // maximize to width and height for all rectangles
-        // search for maximums
-        int maxWidth = -1;
-        int maxHeight = -1;
-        for (MergedPanel panel : mergedPanels) {
-            Dimension panelMaxSize = panel.getLargestRect();
-            if (maxWidth < panelMaxSize.width) {
-                maxWidth = panelMaxSize.width;
-            }
-            if (maxHeight < panelMaxSize.height) {
-                maxHeight = panelMaxSize.height;
-            }
-        }
-        // set maximum size to all rectangles
-        if ((maxWidth == -1) || (maxHeight == -1))
-            return;
-        for (MergedPanel mp : mergedPanels) {
-            mp.setAllCropSize(maxWidth, maxHeight);
+        JLabel label = new JLabel(Messages.getString("BrissGUI.passwordPrompt"));
+
+        JPasswordField pass = new JPasswordField();
+
+        panel.add(label);
+        panel.add(pass);
+
+        String[] options = new String[] {"OK", "Cancel"};
+
+        int option = JOptionPane.showOptionDialog(null, panel, "The title",
+                JOptionPane.NO_OPTION, JOptionPane.PLAIN_MESSAGE,
+                null, options, options[1]);
```

Comment:
My preference here would be to use `OK` as the default action so hitting enter will submit the password. When first trying it out I thought the feature didn't work as I was hitting enter after providing the PW and then nothing happened.